File: client/src/models/vacancy/vacancy.create.dto.ts
import { AttendanceMode } from "../common/attendance.enum";
import { LocationDto } from "../common/location.dto";
import { SkillDto } from "../common/skill.dto";
import { Experience } from "./experience.enum";

export interface VacancyCreate {
    title: string;
    description: string;
    salary: number;
    experience: Experience;
    attendanceMode: AttendanceMode;
    locations: LocationDto[];
    skills: SkillDto[];
    categoryId: string;
    companyId: number;
    recruiterId: string;
}

File: JobSearchApp.Core/Models/Vacancies/VacancyCreateDto.cs
using JobSearchApp.Data.Enums;

namespace JobSearchApp.Core.Models.Vacancies;

public class VacancyCreateDto
{
    public string Title { get; set; } = null!;
    public string Description { get; set; } = null!;
    public double Salary { get; set; }
    public AttendanceMode AttendanceMode { get; set; }
    public Experience Experience { get; set; }
    public bool IsActive { get; set; } = true;
    public int RecruiterId { get; set; }
    public int CategoryId { get; set; }
    public int CompanyId { get; set; }
    public List<LocationDto> Locations { get; set; } = [];
    public List<SkillDto> Skills { get; set; } = [];
}